core/wasmcloud-model.smithy

```diff
@@ -113,6 +113,10 @@ structure wasmbus {
     actorReceive: Boolean,
     /// indicates this service's operations are handled by an provider (default false)
     providerReceive: Boolean,
+    /// Binary message protocol version. Defaults to "0" if unset.
+    /// Be aware that changing this value can break binary compatibility unless
+    /// all users of this interface recompile
+    protocol: String,
```

Comment:
If the default is 0, should this be something like a `protocolVersion` in a `u16`? This works fine I think, just curious on the extensibility we gain by using a String

Reply:
with a string, it could be semver, or different flavors like 2a & 2b.
I wasn't really sure and I thought strings would be more flexible given that we might want more dimensions than monotonic integers